src/graph/graphencodecontext.c

```diff
@@ -0,0 +1,162 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "graphencodecontext.h"
+#include "assert.h"
+#include "../util/rmalloc.h"
+#include "../util/rax_extensions.h"
+
+inline GraphEncodeContext *GraphEncodeContext_New() {
+	GraphEncodeContext *ctx = rm_malloc(sizeof(GraphEncodeContext));
+	ctx->keys = raxNew();
+	GraphEncodeContext_Reset(ctx);
+	return ctx;
+}
+
+inline void GraphEncodeContext_Reset(GraphEncodeContext *ctx) {
+	assert(ctx);
+	ctx->phase = RESET;
+	ctx->keys_processed = 0;
+	ctx->processed_nodes = 0;
+	ctx->processed_edges = 0;
+	ctx->processed_deleted_nodes = 0;
+	ctx->processed_deleted_edges = 0;
+	ctx->datablock_iterator = NULL;
+	ctx->current_relation_matrix_id = 0;
+	ctx->matrix_tuple_iterator = NULL;
+}
+
+inline EncodePhase GraphEncodeContext_GetEncodePhase(const GraphEncodeContext *ctx) {
```

Comment:
I'm a bit on the fence about these abstraction layers of setter/getter APIs. Personally, I rather feel like it's all a bit cumbersome and verbose just to pretend we're not C, and it is certainly not something we've uniformly adopted through the project.

That said, I'll cede the point if you and Roi agree this is the right approach! Let me know.